oktaawscli/aws_auth.py

```diff
@@ -149,7 +149,7 @@ def __create_options_from(roles):
 
     def __find_predefiend_role_from(self, roles):
         found_roles = filter(lambda role_tuple: role_tuple.role_arn == self.role, roles)
-        if not found_roles:
+        if not list(found_roles):
             return None
         else:
             return next(found_roles)
```

Comment:
This is not going to work after you empty the generator to make the list in line 152